types/src/transaction/transaction.rs

```diff
@@ -227,6 +275,42 @@ impl Transaction {
                 Some(amount) if amount == 0 => Err(Error::ZeroAmount),
                 _ => Ok(()),
             },
+            Transaction::AssetCompose {
+                inputs,
+                output,
+                ..
+            } => {
+                if inputs.is_empty() {
+                    return Err(Error::EmptyInput)
+                }
+                for input in inputs {
+                    if input.prev_out.amount == 0 {
+                        return Err(Error::ZeroAmount)
+                    }
+                }
+                match output.amount {
+                    Some(amount) if amount == 1 => Ok(()),
+                    _ => Err(Error::ZeroAmount),
```

Comment:
It seems we need a different error message here for a non-zero amount

Reply:
I replaced it with `InvalidOutput`